SPDY/SPDYProtocol.h

```diff
@@ -288,3 +293,22 @@ extern NSString *const SPDYMetadataStreamTxBytesKey;
 @property NSInteger proxyPort;
 
 @end
+
+/**
+  Protocol that may be implemented by the delegate property of NSURLSession.
+ 
+  This will provide additional context for the request, if desired. Implementing
+  it is optional, and only applies for NSURLSession-based requests. All calls 
+  made using the NSOperationQueue set in delegateQueue in NSURLSession, or else 
+  the mainQueue if one is not set.
+*/
+@protocol SPDYURLSessionDelegate <NSObject>
+
+/**
+  Called just before the request is dispatched and provides the SPDYProtocol 
+  instance handling the request.
+*/
+@optional
+- (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didStartLoadingRequest:(NSURLRequest *)request withSPDYProtocol:(SPDYProtocol *)protocol;
```

Comment:
I'm still pondering the implications of exposing an instance of SPDYProtocol to the app. I'm wondering if we shouldn't make a new object to return, like SPDYRequest or SPDYInstance or something, that strictly contains only what the app needs. SPDYProtocol contains a strong internal reference to the SPDYStream right now. Or maybe just return an instance of the metadata in the callback, since that's all that is currently exposed anyway. Thinking out loud right now.